File: Submissions/128_Longest_Consecutive_Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;
        if(nums.size() == 1) return 1;

        std::set<int> sorted;
        int count = 0;
        int result = 0;
        for(auto num : nums){
            sorted.insert(num);
        }
        auto last_num = std::prev(sorted.end());
        for(auto iter = sorted.begin(); iter != last_num; iter++){
            if ((*std::next(iter) - *iter) == 1) ++count;
            else {
                result = std::max(count + 1, result);
                count = 0;
            }
        }

        result = std::max(count + 1, result);
        
        return result;
    }
};

File: Kotlin/128_Longest_Consecutive_Sequence.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        if (nums.isEmpty()) return 0
        
        var result = 1
        var count = 1
        
        nums.sort()
        
        for(i in 0 .. nums.lastIndex-1){
            if(nums[i+1] - nums[i] == 1) {
                ++count
            }else if(nums[i+1] - nums[i] != 0) {
                result = Math.max(result, count)
                count = 1
            }
        }
        
        result = Math.max(result, count)
        
        return result
    }
}